nautobot/extras/management/commands/runjob.py

```diff
@@ -53,25 +64,42 @@ def handle(self, *args, **options):
             request.id = uuid.uuid4()
             request.user = user
 
+        data = {}
+        try:
+            if options.get("data"):
+                data = json.loads(options["data"])
+        except json.decoder.JSONDecodeError as error:
+            raise CommandError(f"Original Error: {str(error)}\n\n invalid json data.\n\n")
+
         job_content_type = get_job_content_type()
 
         # Run the job and create a new JobResult
         self.stdout.write("[{:%H:%M:%S}] Running {}...".format(timezone.now(), job_class.class_path))
 
-        job_result = JobResult.enqueue_job(
-            run_job,
-            job_class.class_path,
-            job_content_type,
-            user,
-            data={},  # TODO: parsing CLI args into a data dictionary is not currently implemented
-            request=copy_safe_request(request) if request else None,
-            commit=options["commit"],
-        )
-
-        # Wait on the job to finish
-        while job_result.status not in JobResultStatusChoices.TERMINAL_STATE_CHOICES:
-            time.sleep(1)
+        if options["testjob"]:
+            self.stdout.write(
+                "The locally ran Job via testjob is intended for testing ONLY!! This is not suitable for production use."
+            )
+            job = Job.objects.get_for_class_path(job_class.class_path)
+            job_result = run_job_for_testing(
+                job, data=data, request=copy_safe_request(request) if request else None, commit=options["commit"]
+            )
             job_result = JobResult.objects.get(pk=job_result.pk)
```

Comment:
You are correct, and Django's `Model.refresh_from_db()` is meant for that. It's a timing issue since you are running the job synchronously, so you want to refresh the object once the job is complete. https://docs.djangoproject.com/en/4.0/ref/models/instances/#django.db.models.Model.refresh_from_db

Reply:
thanks, that worked! I updated the original as well.